File: 908-middle-of-the-linked-list/middle-of-the-linked-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* middleNode(struct ListNode* head) {
    struct ListNode *front = head;
    struct ListNode *rear = head;
    struct ListNode *next;
    while(front!=NULL && front->next!=NULL)
    {
        rear = rear->next;
        front = front->next->next;
    }
    return rear;
}

File: 908-middle-of-the-linked-list/middle-of-the-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode middleNode(ListNode head) {
        ListNode front = head;
        ListNode rear = head;
        ListNode next;
        while(front!=null && front.next!=null)
        {
            rear = rear.next;
            front = front.next.next;
        }
        return rear;
    }
}